test/Microsoft.Sbom.Targets.Tests/AbstractGenerateSbomTaskTests.cs

```diff
@@ -0,0 +1,264 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Sbom.Api.Utils;
+using Microsoft.Sbom.Contracts;
+using Microsoft.Sbom.Extensions.DependencyInjection;
+using Microsoft.Sbom.Targets;
+using Microsoft.Sbom.Targets.Tests.Utility;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Newtonsoft.Json;
+
+namespace Microsoft.Sbom.Targets.Tests;
+
+/// <summary>
+/// Base class for testing SBOM generation through the GenerateSbomTask.
+/// </summary>
+[TestClass]
+public abstract class AbstractGenerateSbomTaskTests
```

Comment:
Unblocking question, does this class need to be abstract? Is this able to be a regular parent class, as there are no abstract methods defined? Just curious as I'm not too experienced with designing unit test classes 😊

Reply:
When it's abstract, it cannot be instantiated, it needs to first be implemented and then instantiated. Hence, the tests won't run twice. However, good point that it doesn't have abstract methods. I will add the SBOM Specification as an abstract property / getter, so that it's clearer what's needed to be implemented :)

Reply:
Changed at 5cbefd124c2942628989fb2cc697cf6438e46c2b